spec/api.md

```diff
@@ -0,0 +1,129 @@
+# CAS API Documentation
+
+This document describes the HTTP API endpoints used by the CAS (Content Addressable Storage) client to interact with the remote CAS server.
+
+## Authentication
+
+- TODO
+
+## Endpoints
+
+### 1. Get File Reconstruction
+
+**Description**: Retrieves reconstruction information for a specific file, optionally with byte range support.
+
+- **Path**: `/v1/reconstruction/{file_id}`
+- **Method**: `GET`
+- **Parameters**:
+  - `file_id`: MerkleHash in hex format
+- **Headers**:
+  - `Range`: Optional. Format: `bytes={start}-{end}` (inclusive end)
+- **Body**: None
+- **Response**: JSON (`QueryReconstructionResponse`)
+
+  ```json
+  {
+    "offset_into_first_range": 0,
+    "terms": [...],
+    "fetch_info": {...}
+  }
+  ```
+
+- **Error Responses**:
+  - 400 bad request
+  - 404 File not found
+  - 416 Range Not Satisfiable: When requested byte range start exceeds the end of a file
+
+### 2. Batch Get Reconstruction
+
+**Description**: Retrieves reconstruction information for multiple files in a single request.
+
+- **Path**: `/v1/reconstructions`
+- **Method**: `GET`
+- **Query Parameters**:
+  - `file_id`: MerkleHash(es) in hex format (can be repeated)
+  - Example: `/v1/reconstructions?file_id=abc123&file_id=def456`
+- **Headers**: None (beyond authentication)
+- **Body**: None
+- **Response**: JSON (`BatchQueryReconstructionResponse`)
+
+- **Error Responses**:
+  - 400 bad request
+  - 404 File not found if any file is not found
+
+### 3. Query Chunk Deduplication (Global Deduplication)
+
+**Description**: Checks if a chunk exists in the CAS for deduplication purposes.
+
+- **Path**: `/v1/chunk/default-merkledb/{hash}`
+- **Method**: `GET`
+- **Parameters**:
+  - `hash`: Chunk hash in hex format. Review [how to compute chunk hash](spec/hashing.md#Chunk%20Hashes) to compute chunk hashes
+- **Headers**: None (beyond authentication)
+- **Body**: None
+- **Response**: Raw bytes in shard format (chunk data if exists)
+- **Error Responses**:
+  - 404 - Chunk not already tracked by global deduplication
+
+### 4. Upload XORB
+
+**Description**: Uploads a serialized CAS object (XORB) to the server with progress tracking. Review [how to compute xorb hash](spec/hashing.md#Xorb%20Hashes) to compute xorb hashes.
+
+- **Path**: `/v1/xorb/default/{hash}`
```

Comment:
What purpose does `default` serve here? If this were more RESTful, I would expect the path to be `/v1/xorbs/{hash}` where default is an optional parameter.